### PR TITLE
Expose `LanguageInfo` with all Linguist data

### PR DESCRIPTION
**Pull request by look**
As discussed in https://github.com/go-enry/go-enry/issues/54, this provides an API for accessing a `LanguageInfo` struct which is populated with all the data from the Linguist YAML source file. Functions are provided to access the `LanguageInfo` by name or ID.

The other top-level functions like `GetLanguageExtensions`, `GetLanguageGroup`, etc. could in principle be implemented using this structure, which would simplify the code generation. But that would be a big change so I didn't do any of that. Perhaps in the next major version something like that would make sense.

cc @tclem

Closes https://github.com/go-enry/go-enry/issues/54

### CODE REVIEW

File: internal/code-generator/main.go [hunk 1]
```diff
@@ -124,11 +129,12 @@ func main() {
 		{generator.Colors, languagesYAML, "", colorsFile, colorsTmplPath, colorsTmpl, commit},
 		{generator.Groups, languagesYAML, "", groupsFile, groupsTmplPath, groupsTmpl, commit},
 		{generator.ID, languagesYAML, "", idFile, idTmplPath, idTmpl, commit},
+		{generator.LanguageInfo, languagesYAML, "", languageInfoFile, langaugeInfoTmplPath, langaugeInfoTmpl, commit},
 	}
 
 	for _, file := range fileList {
 		if err := file.generate(file.fileToParse, file.samplesDir, file.outPath, file.tmplPath, file.tmplName, file.commit); err != nil {
-			log.Println(err)
+			log.Printf("Error generating template %q to %q: %+v", file.tmplPath, file.outPath, err)
```
Comment (tclem) [inline internal/code-generator/main.go:137]: **Nit:** Maybe keep the newline at the end of this printf?

File: internal/code-generator/assets/languageInfo.go.tmpl [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  MimeType       string
+  TMScope        string
+  AceMode        string
+  CodemirrorMode string
+  Wrap           bool
+  LanguageID     int
+}
+
+// LanguageInfoByName allows accessing LanguageInfo by a language's primary name.
+var LanguageInfoByName = map[string]LanguageInfo{
```
[the remark this answers is not in the document]
Comment (lafriks) [reply internal/code-generator/assets/languageInfo.go.tmpl:24]: My proposal was to just remove name -> info mapping so your use case to get by ID is not affected in any way. As there is already mapping for name -> ID it's possible to just use it and then return info from ID -> info map

File: internal/code-generator/main.go (hunk 1 above)
Comment (lafriks) [inline internal/code-generator/main.go:137]: Small nit:
```suggestion
			log.Printf("error generating template %q to %q: %+v", file.tmplPath, file.outPath, err)
```

**Comment by look** [comment]
@bzz do you mind merging this? I'd like to use this API in our code.

**Pull request merged**